server/src/main/java/org/opensearch/indices/IndicesModule.java

```diff
@@ -174,6 +175,7 @@ public static Map<String, Mapper.TypeParser> getMappers(List<MapperPlugin> mappe
         mappers.put(ConstantKeywordFieldMapper.CONTENT_TYPE, new ConstantKeywordFieldMapper.TypeParser());
         mappers.put(DerivedFieldMapper.CONTENT_TYPE, DerivedFieldMapper.PARSER);
         mappers.put(WildcardFieldMapper.CONTENT_TYPE, WildcardFieldMapper.PARSER);
+        mappers.put(StarTreeMapper.CONTENT_TYPE, new StarTreeMapper.TypeParser());
```

Comment:
Can we add tests to ensure the mapping is parsed only if provided within the expected hierarchy?

Reply:
This test [file](https://github.com/opensearch-project/OpenSearch/pull/14261/files/9560d1951b1358290f39a302801954df2ef7e3b5#diff-0837630e81d5f18d357e692e6455f5fb101e72074541cb4a1aeb4ca57d111337R102) has many such validations. 

I'll add integ tests for tests related to doc values/field data.

Let me know if I'm missing anything specific.